### PR TITLE
Add NXP HVQFN-24 (SOT616-3)

### PR DESCRIPTION
**Pull request by kacper-ka**
This merge updates [hvqfn.yaml](https://github.com/pointhi/kicad-footprint-generator/blob/master/scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml) file and adds `HVQFN-24-1EP_4x4mm_P0.5mm_EP2.5x2.5mm` description.  
Alternate package names:
-  as used by NXP: `SOT616-3`,
- JEDEC code: `MO-220`

#### Package information file
https://www.nxp.com/docs/en/package-information/SOT616-3.pdf

#### Related PRs
KiCad/kicad-symbols#2316
KiCad/kicad-footprints#2060

----

This branch has been updated to include changes in `master` branch and introduced by #494 (among others)

### CODE REVIEW

**Comment by chschlue** [comment]
Pending #422

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 1]
```diff
@@ -392,6 +392,63 @@ VQFN-24-1EP_4x4mm_P0.5mm:
...
+  body_size_x:
+    minimum: 3.9
+    nominal: 4.0
+    maximum: 4.1
+  body_size_y:
+    minimum: 3.9
+    nominal: 4.0
+    maximum: 4.1
+  overall_height:
+    maximum: 1
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: Please add nominal height.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 2]
```diff
@@ -392,6 +392,63 @@ VQFN-24-1EP_4x4mm_P0.5mm:
...
+    maximum: 0.5
+
+  EP_size_x:
+    minimum: 2.45
+    nominal: 2.5
+    maximum: 2.75
+  EP_size_y:
+    minimum: 2.45
+    nominal: 2.5
+    maximum: 2.75
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: Please use package pad dimensions as given in the datasheet and use EP_size_[xy]_overwrite to get a different FP pad size if necessary.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml (hunk 1 above)
Comment (kacper-ka) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: Done.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml (hunk 2 above)
Comment (kacper-ka) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: I don't fully understand your request. EP size is already specified as given in the package datasheet (min 2.45mm, max 2.75mm), nominal size was removed (as not given in the datasheet) but it was derived from the min and max values and conformed with the nominal pad size (2.5x2.5mm) given in the datasheet (fig. 2, page 3).

**Comment by evanshultz** [comment]
@kacper-ka 
The footprint generator uses IPC 7351 rules to generate the footprint. So we need to capture package information, and we ignore the suggested footprint. However, sometimes the EP size needs to vary and so the parameter mentioned above allows an EP size which does not correlate with the pad on the package.

**Comment by kacper-ka** [comment]
@evanshultz, @chschlue - please check, whether I have understood your request correctly.

**Comment by chschlue** [comment]
Yes, looks good to me.

**Comment by chschlue** [comment]
FP merged

**Comment by evanshultz** [comment]
Thank you!

**Pull request merged**